capella_model_explorer/__main__.py

```diff
@@ -0,0 +1,351 @@
+    print(shlex.join(cmd))
+    subprocess.check_call(
+        cmd,
+        env=os.environ
+        | {
+            "CME_LIVE_MODE": "1" if c.LIVE_MODE else "0",
+            "CME_MODEL": c.MODEL,
+            "CME_ROUTE_PREFIX": c.ROUTE_PREFIX,
+            "CME_TEMPLATES_DIR": str(c.TEMPLATES_DIR),
+        },
+    )
+
+
+def run_local_dev() -> None:
+    print("Running the application locally with full reload...")
+    if not pathlib.Path(c.TEMPLATES_DIR).is_dir():
+        raise SystemExit(f"Templates directory not found: {c.TEMPLATES_DIR}")
+    subprocess.check_output(["uvicorn", "--version"])
+    print("Running the application locally...")
+    uvicorn_cmd = [
+        "uvicorn",
+        "--host",
+        c.HOST,
+        "--port",
+        str(c.PORT),
+        "--reload",
+        "--reload-exclude",
+        "git_askpass.py",
+        "--reload-exclude",
+        "input.css",
+        "--reload-dir",
+        ".",
+        "--reload-include",
+        "*.css",
+        "--reload-include",
+        "*.j2",
+        "--reload-include",
+        "*.js",
+        "--reload-include",
+        "*.py",
+        "capella_model_explorer.app:app",
+    ]
+    print(" ".join(uvicorn_cmd))
+    tailwind_proc = build_css(watch=True)
+    assert tailwind_proc is not None
+    time.sleep(1)  # avoid direct uvicorn reload when css file is written
+    uvicorn_proc = subprocess.Popen(
+        uvicorn_cmd,
+        env=os.environ
+        | {
+            "CME_LIVE_MODE": "1",
+            "CME_MODEL": c.MODEL,
+            "CME_TEMPLATES_DIR": c.TEMPLATES_DIR,
+        },
+    )
+    if tailwind_proc is not None:
+        try:
+            with tailwind_proc, uvicorn_proc:
+                while True:
+                    time.sleep(3600)
+        except KeyboardInterrupt:
+            tailwind_proc.terminate()
+            uvicorn_proc.terminate()
+
+
+def build_css(*, watch: bool) -> subprocess.Popen | None:
+    """Build style sheet using Tailwind CSS."""
+    for file in pathlib.Path().glob("static/css/main-*.css"):
+        file.unlink()
+    _install_npm_pkgs()
+    exe = "node_modules/.bin/tailwindcss"
+    subprocess.check_output([exe, "--help"])
+    print("Building style sheet...")
+    input_css = pathlib.Path("static/css/input.css")
+    if not input_css.is_file():
+        raise SystemExit(f"Input CSS file not found: {input_css}")
+    tailwind_cmd = [
+        exe,
+        "--minify",
+        "--input",
+        str(input_css),
+        "--output",
+        f"static/css/main-{capella_model_explorer.__version__}.min.css",
+    ]
+    if watch:
+        tailwind_cmd.append("--watch")
+        print(shlex.join(tailwind_cmd))
+        return subprocess.Popen(tailwind_cmd)
+    print(shlex.join(tailwind_cmd))
+    subprocess.check_call(tailwind_cmd)
+    return None
+
+
+def build_image():
+    if pathlib.Path.cwd().name != "capella-model-explorer":
+        raise SystemExit("Run this command from the root of the project.")
```

Comment:
This condition makes an assumption about the _outside_ of the project, which we have no influence over. This breaks for example if you have a directory structure like `.../capella/model-explorer`.

If you want to check for the repo root, check if `.git/` and `capella_model_explorer/` are present instead.